base/compiler/tfuncs.jl

```diff
@@ -2245,10 +2245,10 @@ function return_type_tfunc(interp::AbstractInterpreter, argtypes::Vector{Any}, s
                     if isa(sv, InferenceState)
                         old_restrict = sv.restrict_abstract_call_sites
                         sv.restrict_abstract_call_sites = false
-                        call = abstract_call(interp, ArgInfo(nothing, argtypes_vec), sv, -1)
+                        call = abstract_call(interp, ArgInfo(nothing, argtypes_vec), sv)
                         sv.restrict_abstract_call_sites = old_restrict
                     else
-                        call = abstract_call(interp, ArgInfo(nothing, argtypes_vec), sv, -1)
+                        call = abstract_call(interp, ArgInfo(nothing, argtypes_vec), sv)
```

Comment:
This needs to match the implementation of return_type https://github.com/JuliaLang/julia/blob/997e3369d2d7454fac22b5c21b7860c4ba350cff/base/compiler/typeinfer.jl#L1142

Reply:
Okay, I synced the implementations of `Core.Compiler.return_type` and `return_type_tfunc`.
One thing I'm not satisfied with is that the current `Core.Compiler.return_type` interface doesn't allow it to see caller module context, so it can't respect module-wise `max_methods` setting.